src/common/server.ts

```diff
@@ -277,6 +255,129 @@ async function createServer(
+      outputChannel.show();
+    }
+  });
+}
+
+async function legacyServerSettingsWarning(settings: string[], outputChannel: LogOutputChannel) {
+  await showWarningMessageWithLogs(
+    "Unsupported settings used with the native server. Refer to the logs for more details.",
+    outputChannel,
+  );
+  traceWarn(
+    `Following settings are not supported with the native server: ${JSON.stringify(settings)}`,
+  );
+}
+
+async function nativeServerSettingsWarning(settings: string[], outputChannel: LogOutputChannel) {
+  await showWarningMessageWithLogs(
+    "Unsupported settings used with the legacy server (ruff-lsp). Refer to the logs for more details.",
+    outputChannel,
+  );
+  traceWarn(
+    `Following settings are not supported with the legacy server (ruff-lsp): ${JSON.stringify(
+      settings,
+    )}`,
+  );
+}
+
+type RuffExecutable = {
+  path: string;
+  version: VersionInfo;
+};
+
+async function resolveNativeServerSetting(
+  settings: ISettings,
+  workspace: vscode.WorkspaceFolder,
+  serverId: string,
+  outputChannel: LogOutputChannel,
+): Promise<{ useNativeServer: boolean; executable: RuffExecutable | undefined }> {
+  let useNativeServer: boolean;
+  let executable: RuffExecutable | undefined;
+
+  switch (settings.nativeServer) {
+    case "on":
+    case true:
+      const legacyServerSettings = getUserSetLegacyServerSettings(serverId, workspace);
+      if (legacyServerSettings.length > 0) {
+        await legacyServerSettingsWarning(legacyServerSettings, outputChannel);
+      }
+      return { useNativeServer: true, executable };
+    case "off":
+    case false:
+      let nativeServerSettings = getUserSetNativeServerSettings(serverId, workspace);
+      if (nativeServerSettings.length > 0) {
+        await nativeServerSettingsWarning(nativeServerSettings, outputChannel);
+      }
+      return { useNativeServer: false, executable };
+    case "auto":
+      const ruffBinaryPath = await findRuffBinaryPath(settings, outputChannel);
+      const ruffVersion = await getRuffVersion(ruffBinaryPath);
+
+      if (supportsNativeServerStable(ruffVersion)) {
+        const legacyServerSettings = getUserSetLegacyServerSettings(serverId, workspace);
+        if (legacyServerSettings.length > 0) {
+          traceInfo(`Legacy server settings found: ${JSON.stringify(legacyServerSettings)}`);
+          useNativeServer = false;
+        } else {
+          useNativeServer = true;
+        }
+      } else {
+        traceInfo(
+          `Stable version of the native server requires Ruff ${versionToString(
+            NATIVE_SERVER_STABLE_VERSION,
+          )}, but found ${versionToString(ruffVersion)} at ${ruffBinaryPath} instead`,
+        );
+        let nativeServerSettings = getUserSetNativeServerSettings(serverId, workspace);
+        if (nativeServerSettings.length > 0) {
+          traceInfo(`Native server settings found: ${JSON.stringify(nativeServerSettings)}`);
+        }
+        useNativeServer = false;
```

Comment:
I think we should at least show a warning in this csae to let users know that one of their settings was ignored.

Reply:
Should we provide a warning for the other case as well where the Ruff version supports the stable server but the user has set legacy server settings?

Reply:
Added a warning with a suggestion for this case:
```
2024-07-08 15:59:18.068 [info] Using the Ruff binary: /Users/dhruv/.local/bin/ruff
2024-07-08 15:59:18.075 [info] Stable version of the native server requires Ruff 0.5.0, but found 0.4.10 at /Users/dhruv/.local/bin/ruff instead
2024-07-08 15:59:21.067 [warning] The following settings are not supported with the legacy server (ruff-lsp): ["ruff.configuration"]
2024-07-08 15:59:21.067 [warning] Please remove these settings or set 'nativeServer' to 'on' to use the native server
2024-07-08 15:59:21.067 [info] Resolved 'ruff.nativeServer: auto' to use the legacy (ruff-lsp) server
```